### ISSUE TITLE
process continually exiting

### ISSUE THREAD

**Issue by BryanHunt** [Enalean/docker-tuleap-aio#14]
I'm running on OS X and I'm continually getting:

2016-06-23 14:25:58,199 INFO success: postfix entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
2016-06-23 14:25:58,202 INFO exited: postfix (exit status 1; not expected)
2016-06-23 14:25:58,202 INFO received SIGCLD indicating a child quit
2016-06-23 14:25:59,206 INFO spawned: 'postfix' with pid 2054
2016-06-23 14:26:00,209 INFO success: postfix entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
2016-06-23 14:26:00,211 INFO exited: postfix (exit status 1; not expected)
2016-06-23 14:26:00,211 INFO received SIGCLD indicating a child quit
2016-06-23 14:26:01,214 INFO spawned: 'postfix' with pid 2055
2016-06-23 14:26:02,216 INFO success: postfix entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)

**Comment by LeSuisse** [comment]
Hello,

Could please the logs in /var/log/maillog inside the container to see if we can have more information?

**Comment by BryanHunt** [comment]
Jun 23 14:43:16 2f34db3b5e20 postfix[3609]: fatal: bad string length 0 < 1: myhostname = 
Jun 23 14:43:18 2f34db3b5e20 postfix[3610]: fatal: bad string length 0 < 1: myhostname = 
Jun 23 14:43:20 2f34db3b5e20 postfix[3611]: fatal: bad string length 0 < 1: myhostname = 
Jun 23 14:43:22 2f34db3b5e20 postfix[3612]: fatal: bad string length 0 < 1: myhostname = 
Jun 23 14:43:24 2f34db3b5e20 postfix[3613]: fatal: bad string length 0 < 1: myhostname =

**Comment by LeSuisse** [comment]
Have you properly set the environment variable VIRTUAL_HOST on the first run?

**Comment by BryanHunt** [comment]
I did the initial run using the readme ...
`docker run -ti -e VIRTUAL_HOST=localhost -p 80:80 -p 443:443 -p 22:22 -v tuleap-data:/data enalean/tuleap-aio`
Subsequent runs have been from Kitematic.

**Comment by LeSuisse** [comment]
I'm not able to reproduce, could you please try again with a new data container?

**Comment by nsshenoy** [comment]
I have windows 8 machine. 
in Kitematic searched for tuleap container. downloaded and tried to start tuleap. getting error

==============================================
Installation completed successfully!
- WHAT TO DO TO FINISH THE TULEAP INSTALLATION (see /root/todo_tuleap.txt)
- Your domain name is 
- Automatically generated passwords are stored in /data/root/.tuleap_passwd
- Customize /etc/tuleap/conf/local.inc and /etc/tuleap/conf/database.inc
- You may also want to customize /etc/httpd/conf/httpd.conf
- Customize /etc/tuleap/site-content information for your site.
-   For instance: contact/contact.txt 
- 
- Default admin credentials are login: admin / password: tVJFjoPdxgQjjNG
- CHANGE DEFAULT CREDENTIALS BEFORE FIRST USAGE
- Auto generated passwords (mysql, application, etc) are stored in /data/root/.tuleap_passwd
- 
- -----------------------------------------
- This TODO list is available in /root/todo_tuleap.txt.
Stopping mysqld:  G[  OK  ]
Stopping httpd: G[  OK  ]
Stopping crond: G[  OK  ]
+ ./boot-fixpath.sh
+ ./boot-update-config.sh
+ perl -pi -e 's%^short_open_tag = Off%short_open_tag = On%' /etc/php.ini
+ perl -pi -e 's%^;date.timezone =%date.timezone = Europe/Paris%' /etc/php.ini
+ perl -pi -e 's%^#myhostname = host.domain.tld%myhostname = %' /etc/postfix/main.cf
+ perl -pi -e 's%^alias_maps = hash:/etc/aliases%alias_maps = hash:/etc/aliases,hash:/etc/aliases.codendi%' /etc/postfix/main.cf
+ perl -pi -e 's%^alias_database = hash:/etc/aliases%alias_database = hash:/etc/aliases,hash:/etc/aliases.codendi%' /etc/postfix/main.cf
+ perl -pi -e 's%^#recipient_delimiter = %recipient_delimiter = %' /etc/postfix/main.cf
+ perl -pi -e 's%enable-cache[\t ]+group[\t ]+yes%enable-cache group no%' /etc/nscd.conf
+ source mysql-utils.sh
++ set -e
+ start_mysql
+ echo 'Start mysql'
Start mysql
+ sleep 1
+ /usr/bin/python /usr/lib/python2.6/site-packages/supervisor/pidproxy.py /var/run/mysqld/mysqld.pid /usr/bin/mysqld_safe
160722 06:36:22 mysqld_safe Logging to '/var/lib/mysql/57a4fb4c57d1.err'.
160722 06:36:22 mysqld_safe Starting mysqld daemon with databases from /var/lib/mysql
++ sed -e 's/^\$sys_dbpasswd="\(.*\)";$/\1/'
++ egrep '^\$sys_dbpasswd' /etc/tuleap/conf/database.inc
+ mysql -ucodendiadm -p7dfQKSiA9s7RTFh -e 'show databases'
+ '[' true == false ']'
+ /etc/init.d/tuleap start
Restore application cache directories:
info Restore ownership to /var/tmp/tuleap_cache
info Restore ownership to /usr/share/tuleap/src/www/scripts/combined
info Restore ownership to /var/tmp/tuleap_cache/lang
info Restore ownership to /var/tmp/tuleap_cache/hooks.json
Invalidate application caches:
info Invalidate Restler cache
info Invalidate language cache
info Invalidate JS combined cache
info Invalidate WSDL
info Invalidate Plugin hooks
Starting Tuleap: [OK]
+ stop_mysql
+ echo 'Stop mysql'
Stop mysql
++ cat /var/run/mysqld/mysqld.pid
+ PID=1131
+ kill -15 1131
+ ps -p 1131
+ echo 'Waiting for mysql (1131) to stop'
Waiting for mysql (1131) to stop
+ sleep 1
+ ps -p 1131
+ echo 'Waiting for mysql (1131) to stop'
Waiting for mysql (1131) to stop
+ sleep 1
+ ps -p 1131
+ echo 'Waiting for mysql (1131) to stop'
Waiting for mysql (1131) to stop
+ sleep 1
+ ps -p 1131
+ echo 'Waiting for mysql (1131) to stop'
Waiting for mysql (1131) to stop
+ sleep 1
+ ps -p 1131
+ echo 'Waiting for mysql (1131) to stop'
Waiting for mysql (1131) to stop
+ sleep 1
160722 06:36:28 mysqld_safe mysqld from pid file /var/run/mysqld/mysqld.pid ended
+ ps -p 1131
+ exec supervisord -n
2016-07-22 06:36:28,983 CRIT Supervisor running as root (no user in config file)
2016-07-22 06:36:28,990 INFO supervisord started with pid 1
2016-07-22 06:36:28,991 INFO spawned: 'httpd' with pid 1170
2016-07-22 06:36:28,993 INFO spawned: 'sshd' with pid 1171
2016-07-22 06:36:28,996 INFO spawned: 'postfix' with pid 1172
2016-07-22 06:36:29,007 INFO spawned: 'rsyslog' with pid 1173
2016-07-22 06:36:29,012 INFO spawned: 'mysql' with pid 1174
2016-07-22 06:36:29,029 INFO spawned: 'crond' with pid 1175
2016-07-22 06:36:29,036 INFO success: postfix entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
2016-07-22 06:36:29,076 INFO exited: httpd (exit status 1; not expected)
2016-07-22 06:36:29,078 INFO received SIGCLD indicating a child quit
2016-07-22 06:36:30,040 INFO success: sshd entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2016-07-22 06:36:30,041 INFO success: mysql entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2016-07-22 06:36:30,043 INFO success: crond entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2016-07-22 06:36:30,045 INFO exited: postfix (exit status 1; not expected)
2016-07-22 06:36:30,047 INFO received SIGCLD indicating a child quit
2016-07-22 06:36:31,054 INFO spawned: 'httpd' with pid 1284
2016-07-22 06:36:31,062 INFO spawned: 'postfix' with pid 1285
2016-07-22 06:36:31,126 INFO success: postfix entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
2016-07-22 06:36:31,141 INFO exited: httpd (exit status 1; not expected)
2016-07-22 06:36:31,141 INFO received SIGCLD indicating a child quit
2016-07-22 06:36:32,108 INFO exited: postfix (exit status 1; not expected)
2016-07-22 06:36:32,109 INFO received SIGCLD indicating a child quit
2016-07-22 06:36:33,117 INFO spawned: 'postfix' with pid 1286
2016-07-22 06:36:34,123 INFO success: postfix entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
2016-07-22 06:36:34,124 INFO success: rsyslog entered RUNNING state, process has stayed up for > than 5 seconds (startsecs)
2016-07-22 06:36:34,130 INFO spawned: 'httpd' with pid 1287
2016-07-22 06:36:34,143 INFO exited: postfix (exit status 1; not expected)
2016-07-22 06:36:34,146 INFO received SIGCLD indicating a child quit
2016-07-22 06:36:34,214 INFO exited: httpd (exit status 1; not expected)
2016-07-22 06:36:34,214 INFO received SIGCLD indicating a child quit
2016-07-22 06:36:35,217 INFO spawned: 'postfix' with pid 1288
2016-07-22 06:36:36,221 INFO success: postfix entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
2016-07-22 06:36:36,231 INFO exited: postfix (exit status 1; not expected)
2016-07-22 06:36:36,233 INFO received SIGCLD indicating a child quit
2016-07-22 06:36:36,396 CRIT received SIGTERM indicating exit request

**Comment by LeSuisse** [comment]
Since the issue seems related to Kitematic you should check if the environment variable VIRTUAL_HOST is properly passed to the container.

For example, something like ``docker run --rm -e VIRTUAL_HOST=tuleap_host alpine sh -c 'echo "$VIRTUAL_HOST"'`` should return the string tuleap_host.

**Comment by tuanha-viosoft** [comment]
Hello,
I have the same issue
`+ exec supervisord -n
2016-08-26 10:59:23,098 CRIT Supervisor running as root (no user in config file)
2016-08-26 10:59:23,105 INFO supervisord started with pid 1
2016-08-26 10:59:23,105 INFO spawned: 'httpd' with pid 217
2016-08-26 10:59:23,106 INFO spawned: 'sshd' with pid 218
2016-08-26 10:59:23,107 INFO spawned: 'postfix' with pid 219
2016-08-26 10:59:23,107 INFO spawned: 'rsyslog' with pid 220
2016-08-26 10:59:23,108 INFO spawned: 'mysql' with pid 221
2016-08-26 10:59:23,109 INFO spawned: 'crond' with pid 222
2016-08-26 10:59:23,111 INFO success: postfix entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
2016-08-26 10:59:23,135 INFO exited: httpd (exit status 1; not expected)
2016-08-26 10:59:23,135 INFO received SIGCLD indicating a child quit
2016-08-26 10:59:23,240 INFO exited: postfix (exit status 0; expected)
2016-08-26 10:59:23,240 INFO received SIGCLD indicating a child quit
2016-08-26 10:59:24,242 INFO success: sshd entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2016-08-26 10:59:24,242 INFO success: mysql entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2016-08-26 10:59:24,243 INFO success: crond entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2016-08-26 10:59:24,244 INFO spawned: 'httpd' with pid 401
2016-08-26 10:59:24,296 INFO exited: httpd (exit status 1; not expected)
2016-08-26 10:59:24,297 INFO received SIGCLD indicating a child quit
2016-08-26 10:59:26,300 INFO spawned: 'httpd' with pid 402
2016-08-26 10:59:26,360 INFO exited: httpd (exit status 1; not expected)
2016-08-26 10:59:26,361 INFO received SIGCLD indicating a child quit
2016-08-26 10:59:28,364 INFO success: rsyslog entered RUNNING state, process has stayed up for > than 5 seconds (startsecs)
2016-08-26 10:59:29,367 INFO spawned: 'httpd' with pid 403
2016-08-26 10:59:29,425 INFO exited: httpd (exit status 1; not expected)
2016-08-26 10:59:29,425 INFO received SIGCLD indicating a child quit
2016-08-26 10:59:30,427 INFO gave up: httpd entered FATAL state, too many start retries too quickly
`

**Comment by LeSuisse** [comment]
Do you also use Kitematic?

**Comment by tuanha-viosoft** [comment]
I follow this guide: https://hub.docker.com/r/enalean/tuleap-aio/
`# docker pull enalean/tuleap-aio
# docker volume create --name tuleap-data
# docker run -ti -e VIRTUAL_HOST=localhost -p 5080:80 -p 5443:443 -p 5022:22 -v tuleap-data:/data enalean/tuleap-aio`

**Comment by LeSuisse** [comment]
That was not the question: do you also use Kitematic?

If no, which version of Docker are you running and how?

I just tested again from scratch with the latest image, I'm still not able to reproduce the issue. I'm starting to think the issue is not related to the tuleap-aio image.

**Comment by tuanha-viosoft** [comment]
No, I don't use Kitematic
I'm using Docker version 1.12.1, build 23cf638
I installed Docker on Ubuntu 14.04 follow this https://docs.docker.com/engine/installation/linux/ubuntulinux/
And then, I downloaded tuleap-aio container https://hub.docker.com/r/enalean/tuleap-aio/

**Comment by LeSuisse** [comment]
Ok, can you give us the result of:
```
docker exec <container_name> cat /var/log/httpd/error_log
```
where <container_name> is the name of the container running your Tuleap instance.

**Comment by tuanha-viosoft** [comment]
Hi LeSuisse,
There is no error_log file in /var/log/http/. It's an empty folder.

**Comment by LeSuisse** [comment]
Hum that's weird. What's the result of:
```
docker exec <container_name> cat /etc/httpd/conf/httpd.conf
```

**Comment by tuanha-viosoft** [comment]
This is the result:

#
# VirtualHost example:
# Almost any Apache directive may go into a VirtualHost container.
# The first VirtualHost section is used for requests without a known
# server name.
#
#<VirtualHost *:80>
#    ServerAdmin webmaster@dummy-host.example.com
#    DocumentRoot /www/docs/dummy-host.example.com
#    ServerName dummy-host.example.com
#    ErrorLog logs/dummy-host.example.com-error_log
#    CustomLog logs/dummy-host.example.com-access_log common
#</VirtualHost>

# Include SSL related configuration
Include conf/ssl.conf

# Load plugins specific apache configuration
Include conf.d/*.conf

DocumentRoot "/usr/share/tuleap/src/www"

<Directory "/usr/share/tuleap/src/www">
    Options Includes FollowSymLinks MultiViews
    AllowOverride All
    AcceptPathInfo On
    Order allow,deny
    Allow from all
    SetEnvIfNoCase Request_URI /file/download\.php no-gzip dont-vary
</Directory>

# Don't allow access to API for anyone.
# Currently, the risk is low, so we only filter on UserAgent string (which can be easily forged)
# For better security, you can filter on the server IP address
SetEnvIf User-Agent ^Codendi let_me_in
<Directory "/usr/share/tuleap/src/www/api/reference">
    # API scrips do not end with ".php"
    ForceType application/x-httpd-php
    Order Deny,Allow
    Deny from all
    Allow from env=let_me_in
</Directory>

#
# On the production server the local.inc file is in /etc/tuleap/conf/local.inc
# because it contains sensitive data
#
SetEnv TULEAP_LOCAL_INC ""

**Comment by LeSuisse** [comment]
Either this is just a part of the file or something went really wrong at some point. At this point the easiest way is just to trash the data container and start over. While you are at it verify that you the latest image.

Your issue is not related with the original issue of this thread.

And please when you copy/paste config file use a code block, it is unreadable otherwise.

**Comment by LeSuisse** [comment]
Since this issue is opened since 20 days without response and I'm not able to reproduce the initial problem I close it.
Feel free to open a new one if needed.

**Comment by geektortoise** [comment]
Trashing the data container worked for me.

Thanks !

**Comment by seyedmmousavi** [comment]
Is there any solution?

**Comment by LeSuisse** [comment]
This issue is 6 months old, if you are still using a 6 months old image you should upgrade, if that's not the case please open an new issue since your issue is probably not related to this one.

**Comment by seyedmmousavi** [comment]
OK, thanks
yes I have same issue with current version.

**Comment by rgstephens** [comment]
I'm running into the same issue on Ubuntu, latest version (4 days old).

**Comment by LeSuisse** [comment]
As said before, please open a new issue. It would help if you attach to it, at the bare minimum, the container logs and if the issue is Postfix related, it would be nice if you also attach the content of your `/var/log/maillog` inside your container.

**Comment by lounice** [comment]
I had the same error and solved it by defining the working_dir: "/root/app" in docker-compose

**Comment by LeSuisse** [comment]
Hi,

This issue is becoming the go to for every issues where the container exit with an error and that's not related to the initial report anymore. For information an issue with Postfix has been fixed in #30.
I'm going to lock this conversation, please open a dedicated issue with enough information so it can be investigated (and fixed if needed) if you have a similar problem.
